test/studies/shootout/binary-trees/binarytrees-pool.chpl

```diff
@@ -0,0 +1,93 @@
+/* The Computer Language Benchmarks Game
+   https://salsa.debian.org/benchmarksgame-team/benchmarksgame/
+
+   contributed by Casey Battaglino, Ben Harshbarger, Brad Chamberlain, and Jade Abraham
+   derived from the GNU C version by Jeremy Zerfas
+*/
+
+use Allocators;
+
+config const n = 10;         // the maximum tree depth
+config const globalPoolSize = 2 ** 32;
+config const localPoolSize = 2 ** 32;
+
+var globalPool = new bumpPtrMemPool(globalPoolSize, parSafe=false);
+
+proc main() {
+  const minDepth = 4,                      // the shallowest tree
+        maxDepth = max(minDepth + 2, n),   // the deepest normal tree
+        strDepth = maxDepth + 1,           // the depth of the "stretch" tree
+        depths = minDepth..maxDepth by 2;  // the range of depths to create
+  var stats: [depths] (int,int);           // stores statistics for the trees
+
+  //
+  // Create the short-lived "stretch" tree, checksum it, and print its stats.
+  //
+  {
+    const strTree = __primitive("new with allocator", globalPool, unmanaged Tree, strDepth, globalPool);
```

Comment:
This is currently one of my blockers on finishing this PR, using `newWithAllocator` here doesn't work, because `globalPool` is passed by `ref` to the initializer.

See [test/library/standard/Allocators/refTest.chpl](https://github.com/chapel-lang/chapel/pull/25412/files#diff-785fa926a69f4cf4319f580212df3af120e2a2f56c99aa7c0c94add5ec510780) for a smaller example of the issue.

Reply:
Something about that comment doesn't make sense. Did you mean `const ref` (vs `ref`?)

Anyway, IMO it'd be reasonable for allocators to be class instances. Would that help?

Reply:
I did mean `ref`, the `allocate` call modifies the internal state of the custom allocator, so it must be a modifiable reference.

I will try classes, I think that would solve this issue

Reply:
Making it a class solved this issue 🎉